Client/Frontend/Browser/New Tab Page/NewTabPageViewController.swift

```diff
@@ -424,8 +455,167 @@ class NewTabPageViewController: UIViewController, Themeable {
+                feedOverlayView.loaderView.start()
+                
+                if let section = layout.braveTodaySection {
+                    let numberOfItems = collectionView.numberOfItems(inSection: section)
+                    if numberOfItems > 0 {
+                        collectionView.deleteItems(
+                            at: (0..<numberOfItems).map({ IndexPath(item: $0, section: section) })
+                        )
+                    }
+                }
+            }
+        case (.loading, _):
+            UIView.animate(withDuration: 0.2, animations: {
+                self.feedOverlayView.loaderView.alpha = 0.0
+            }, completion: { _ in
+                self.feedOverlayView.loaderView.stop()
+                self.feedOverlayView.loaderView.alpha = 1.0
+                self.feedOverlayView.loaderView.isHidden = true
+            })
+            if collectionView.contentOffset.y == collectionView.contentInset.top {
+                collectionView.reloadData()
+                collectionView.layoutIfNeeded()
+                let cells = collectionView.indexPathsForVisibleItems
+                    .filter { $0.section == section }
+                    .compactMap(collectionView.cellForItem(at:))
+                cells.forEach { cell in
+                    cell.transform = .init(translationX: 0, y: 200)
+                    UIView.animate(withDuration: 0.5, delay: 0, usingSpringWithDamping: 1.0, initialSpringVelocity: 0, options: [.beginFromCurrentState], animations: {
+                        cell.transform = .identity
+                    }, completion: nil)
+                }
+            } else {
+                collectionView.reloadSections(IndexSet(integer: section))
+            }
+        default:
+            collectionView.reloadSections(IndexSet(integer: section))
+        }
+    }
+    
+    @objc private func checkForUpdatedFeed() {
+        if isBraveTodayVisible {
+            if collectionView.contentOffset.y == collectionView.contentInset.top {
+                // Reload contents if the user is not currently scrolled into the feed
+                loadFeedContents()
+            } else {
+                // Possibly show the "new content available" button
+                if feedDataSource.shouldLoadContent {
+                    feedOverlayView.showNewContentAvailableButton()
+                }
+            }
+        }
+    }
+    
+    private func loadFeedContents(completion: (() -> Void)? = nil) {
+        if !feedDataSource.shouldLoadContent {
+            return
+        }
+        feedDataSource.load(completion)
+    }
+    
     // MARK: - Actions
     
+    @objc private func tappedNewContentAvailable() {
+        if case .loading = feedDataSource.state {
+            return
+        }
+        let todayStart = collectionView.frame.height - feedOverlayView.headerView.bounds.height - 32 - 16
+        newContentAvailableDismissTimer = nil
```

Comment:
doesn't this timer need to be invalidated before setting it to nil? perhaps it causes the retain cycle

Reply:
`newContentAvailableDismissTimer` has a `didSet` which invalidates on setting a new value